### PR TITLE
Added transformer-xl hugging face

### PR DESCRIPTION
**Pull request by darius-lam**

### CODE REVIEW

File: transformer_xl/data_loader.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+""" Stolen from https://github.com/nshepperd/gpt-2/blob/finetuning/src/load_dataset.py"""
```
Comment (8enmann) [inline transformer_xl/data_loader.py:1]: Odd to use the GPT-2 tokenizer on TXL, any reason for this?

**Comment by 8enmann** [comment]
Looks ok, any chance you could reuse more code? How were the results?

File: transformer_xl/data_loader.py (hunk 1 above)
Comment (8enmann) [reply transformer_xl/data_loader.py:1]: Ah, now I see it's copy-pasta and you're passing in a different encoder. Can you either remove the __main__ below or update it? Also probably want to change the hash so it doesn't collide with cached files from GPT-2 encoding.

Comment (darius-lam) [reply transformer_xl/data_loader.py:1]: Oops, that’s a typo— the script itself has been modified to support the transformer-xl tokenizer. When loaded into train.py the script runs fine; I will fix up main() and re-submit.

Comment (darius-lam) [reply transformer_xl/data_loader.py:1]: ^^ Yep, I’ll change the hash as well

**Comment by darius-lam** [comment]
It’s able to memorize a tiny subset of the Gutenberg Ulysses on a Gtx 1080 Ti relatively quickly.  Am planning to run larger scale experiments soon. Will update.

I’ll also take a look at writing a tokenizer-agnostic data loader; this was mostly a quick run to see if I could get the HuggingFace TXL to train at all. Can definitely reuse more code.

**Comment by 8enmann** [comment]
Great! Happy to let you check it in on the dirty side and fix up later. Let me know when you're ready to merge.

**Pull request merged**